### DISCUSSION TITLE
How to get an array of key-value pairs in a query

### DISCUSSION THREAD

**Discussion by tingfeng-key** [tokio-rs/axum#1482] [Q&A] [answered]
query：
```
pageNum: 1
pageSize: 10
dictName: 1
dictType: 1
params[beginTime]: 2022-10-01
params[endTime]: 2022-10-20
```
How to parse beginTime and endTime

**Comment by davidpdrsn** [comment]
Try https://docs.rs/axum-extra/0.4.0-rc.1/axum_extra/extract/struct.Query.html. Otherwise you can make a Query extractor that uses serde_qs

**Reply by jplatte** [reply, marked as answer]
OP will have to use `serde_qs`, `serde_html_form` which the `axum-extra` extractor is based on doesn't really allow this syntax, unless you want a flat list of fields in which case `serde(rename)` works with `axum`s `Query` extractor as well:

```rust
#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
struct QueryParameters {
    page_num: u32,
    page_size: u32,
    dict_name: u32,
    dict_type: u32,
    #[serde(rename = "params[beginTime]")]
    begin_time: String,
    #[serde(rename = "params[endTime]")]
    end_time: String,
}
```

**Reply by tingfeng-key** [reply]
Thank you, this way works fine
